packages/blade/src/components/BottomSheet/utils.ts

```diff
@@ -55,4 +56,70 @@ function computeMaxContent(
   );
 }
 
-export { SnapPoints, computeMaxContent, computeMinContent, computeSnapPointBounds };
+type TrailingComponents = 'Button' | 'Badge' | 'Link' | 'Text';
+
+// prop restriction map for corresponding sub components
+const propRestrictionMap = {
+  Button: {
+    size: 'xsmall',
+    variant: 'tertiary',
+  },
+  Badge: {
+    size: 'medium',
+  },
+  Link: {
+    size: 'medium',
+  },
+  Text: {
+    size: 'medium',
+    variant: 'body',
+  },
+} as const;
+
+const useBottomSheetHeaderTrailingRestriction = (trailing: React.ReactNode) => {
+  // Could've used refs, but in react-native for some reason the enhancedTrailing
+  // is changing to null on opening the bottomsheet
+  // thus saving it to state only
+  const [
+    enhancedTrailingComponent,
```

Comment:
what does "enhancedTrailing" mean 🤔 ?

Reply:
So enhancedTrailing is the trailing component with restrictions applied. 

For example, say you pass IconButton the enhancedTrailing will contain `<IconButton variant="tertiary" size="xsmall" />`

Reply:
Done. validatedTrailing.